### ISSUE TITLE
Flag that tells if HU source is CDC

### ISSUE THREAD

**Issue by lucumon** [JohnyMielony/ESP32TLCDCEmu#5]
Hi!

I'm working on a program to display the song title on the AFFA2 ++, similar to [MeganeBT](https://github.com/Tomasz-Mankowski/MeganeBT), using your project. I'm trying to find a flag that indicates if the HU is using the CDC as a source, or if it's on FM/AM. Does such a flag exist in the CDC communication?

Thanks!

**Comment by lucumon** [comment]
I tried setting up your project for the first time, and I'm not getting the changer source on my radio, only fm and am. I tested communications with an oscilloscope and they seem to be talking to each other. Any help? I'll post scope screenshots shortly.

**Comment by JohnyMielony** [comment]
I don't know if there is such flag. If there is, it is not on connection between CDC and HU, but between HU and display. There is no need to CDC know that.
If CDC is not showing as an input on HU that means communications issues between them. Debug frames (not electrical signal itself) to find out if they're exchanged in the right order. If frames are correct and CRC checksum checks out.

**Comment by lucky1804** [comment]
> I tried setting up your project for the first time, and I'm not getting the changer source on my radio, only fm and am. I tested communications with an oscilloscope and they seem to be talking to each other. Any help? I'll post scope screenshots shortly.

@lucumon Were you ever able to do that, I am trying to get something similar like MeganeBT with track info display but MeganeBT uses Bluetooth SoC that is discontinued.

**Comment by lucumon** [comment]
> @lucumon Were you ever able to do that, I am trying to get something similar like MeganeBT with track info display but MeganeBT uses Bluetooth SoC that is discontinued.

No, sorry. After some trial and error, I gave up and ended up purchasing a cheap Android screen from AliExpress. I got mine for €70, perfectly integrated into the dash and with CarPlay/Android Auto.
